examples/module/mpc/linear.py

```diff
@@ -0,0 +1,140 @@
+        current_x[...,0,:] = x_init
+        for i in range(T-1):
+            current_x[...,i+1,:], _ = lti(current_x[...,i,:], current_u[...,i,:])
+        time_expert  = torch.arange(0, T, device=device)
+
+        mpc_expert = pp.module.MPC(lti, T, step=1)
+        x_true, u_true, cost_true = mpc_expert.forward(expert['Q'], expert['p'], x_init, current_x, current_u, time_expert)
+
+        torch.manual_seed(args.seed)
+        A = (torch.eye(n_state, device=device) + 0.2 * torch.randn(n_state, n_state, device=device))\
+            .requires_grad_()
+        B = torch.randn(n_state, n_ctrl, device=device).requires_grad_()
+
+        fname = os.path.join(args.save, 'pypose losses.csv')
+        loss_f = open(fname, 'w')
+        loss_f.write('im_loss,mse\n')
+        loss_f.flush()
+        fname = os.path.join(args.save, 'pypose time.csv')
+        time_d = open(fname, 'w')
+        time_d.write('backward_time,overall_time\n')
+        time_d.flush()
+
+        def get_loss(x_init, _A, _B):
+            lti_ = pp.module.LTI(_A, _B, C, D, c1, c2)
+            current_x_ = torch.zeros(n_batch, T, n_state, device=device)
+            current_u_ = torch.zeros(n_batch, T, n_ctrl, device=device)
+            current_x_[...,0,:] = x_init
+            for i in range(T-1):
+                current_x_[...,i+1,:], _ = lti_(current_x_[...,i,:], current_u_[...,i,:])
+            time_agent  = torch.arange(0, T, device=device)
+            mpc_agent = pp.module.MPC(lti_, T, step=1)
+            x_pred, u_pred, cost_pred = mpc_agent.forward(expert['Q'], expert['p'], x_init, current_x_, current_u_, time_agent)
+
+            traj_loss = torch.mean((u_true - u_pred)**2) \
+                + torch.mean((x_true - x_pred)**2)
+
+            return traj_loss
+
+        opt = optim.RMSprop((A, B), lr=1e-2)
+
+        for i in range(150):
+            t1 = time.time()
+            traj_loss = get_loss(x_init, A, B)
+            opt.zero_grad()
+            t2 = time.time()
+            traj_loss.backward()
+            t3 = time.time()
+            backward_time = t3 - t2
+            opt.step()
+            t4 = time.time()
+            overall_time = t4 - t1
+
+            model_loss = torch.mean((A - expert['A'])**2) + \
+                        torch.mean((B - expert['B'])**2)
+
+            loss_f.write('{},{}\n'.format(traj_loss.item(), model_loss.item()))
+            loss_f.flush()
+            time_d.write('{},{}\n'.format(backward_time, overall_time))
+            time_d.flush()
+
+            plot_interval = 100
+            if i % plot_interval == 0:
+                print(A, expert['A'])
+                print(B, expert['B'])
+
+            print('{:04d}: traj_loss: {:.4f} model_loss: {:.4f}'.format(
+                i, traj_loss.item(), model_loss.item()))
+            print("backward_time = ", backward_time)
+            print("overall_time = ", overall_time)
+
+
+if __name__=='__main__':
+    train = TrainMPC()
```

Comment:
This example doesn't seem to be correct when I run it. The model loss keep increasing.